src/utils.jl

```diff
@@ -315,5 +315,8 @@ function Base.show(io::IO, ::MIME"text/plain", m::SBML.Model)
     )
 end
 
-# Define equality `==` method for `MathApply`.
+# Define equality `==` methods for some of our custom types.
 Base.:(==)(a::MathApply, b::MathApply) = a.fn == b.fn && a.args == b.args
+Base.:(==)(a::AlgebraicRule, b::AlgebraicRule) = a.math == b.math
+Base.:(==)(a::T, b::T) where {T<:Union{AssignmentRule,RateRule}} =
```

Comment:
This can compare AssignmentRule to be equal to RateRule right?

Reply:
(maybe not, I'm not sure if <: destroys the Union or not but I'd assume not.)

Reply:
No, they'd be different:
```julia
julia> SBML.RateRule("x", SBML.MathVal(1)) == SBML.AssignmentRule("x", SBML.MathVal(1))
false

julia> SBML.RateRule("x", SBML.MathVal(1)) == SBML.RateRule("x", SBML.MathVal(1))
true

julia> SBML.AssignmentRule("x", SBML.MathVal(1)) == SBML.AssignmentRule("x", SBML.MathVal(1))
true
```
A concrete object can have either `AssignmentRule` or `RateRule` type, so `T` would be only one type of the union and both arguments must have the same type.

Reply:
ok seems like T is forced to be rank1. :D Solved